### PR TITLE
[PATCH API-NEXT v2] api: comp: compression API set and implementation

### PR DESCRIPTION
**Pull request by 1234sv**
An API set to add compression support in ODP.
Current Patch series divided into

1. user guide and spec
2. linux-generic implementation, and 
3. tests to validate API implementation.

All tests have been passed on Travis, except
1. doxygen warnings raised on some unrelated files and script changes
2. checkpatch.pl execution reports no error still script reporting error.

### CODE REVIEW

**Comment by lumag** [comment]
@1234sv please update first patch to include empty line between subject line and the rest of commit message.

**Comment by 1234sv** [comment]
@Bill-Fischofer-Linaro these are probably raised on 1st patch and are fixed in subsequent commits.

File: .travis.yml [hunk 1]
```diff
@@ -251,11 +251,17 @@ jobs:
...
+                          - |
+                             fgrep -rq warning ./test.log
+                             if [ $? -eq 0 ]; then
+                               false
+                             else
+                               true
+                             fi
                 - stage: test
                   env: TEST=checkpatch
                   compiler: gcc
```
Comment (Bill-Fischofer-Linaro) [inline .travis.yml]: This should not be part of this PR. Neither should the extraneous `.checkpatch-camelcase.git.3c6350f` file.

File: include/odp/api/spec/comp.h [hunk 2]
```diff
@@ -21,8 +21,8 @@
 extern "C" {
```
Comment (Bill-Fischofer-Linaro) [inline include/odp/api/spec/comp.h]: API changes really should be a separate commit.

**Comment by muvarov** [comment]
This will be subject for 2.0 branch.

**Comment by Bill-Fischofer-Linaro** [comment]
@bala-manoharan Is this superseded by PR #580? If so please close this PR.

**Comment by bala-manoharan** [comment]
Superseded by PR 580

**Comment by Bill-Fischofer-Linaro** [comment]
Superseded by PR #580, per @bala-manoharan
